### PR TITLE
Simplify checks for minimum library versions

### PR DESCRIPTION
**Pull request by hughmcmaster**
This patch builds on the previous patch by simplifying the check for minimum library versions.

### CODE REVIEW

File: setupinfo.py [hunk 1]
```diff
@@ -445,6 +434,17 @@ def get_library_versions():
     sys.exit(1)
 
 
+def check_build_dependencies():
+    xml2_version, xslt_version = get_library_versions()
+    if (not check_min_version(xml2_version, '2.7.0', 'libxml2') or
+            not check_min_version(xslt_version, '1.1.23', 'libxslt')):
+        sys.exit(1)
```
Comment (scoder) [inline setupinfo.py]: I know, this is part of a build script, but I'd rather not see this _function_ kill the runtime. A return value is always better for a function like this.

Comment (hughmcmaster) [reply setupinfo.py]: Doesn't that make it possible that the build might fail if a user's version of libxml2 and/or libxslt is too old? That said, it's probably unlikely, as those versions have been available for several years now.

Would you be happy with returning True/False, then from the caller, raising an exception? I'll push that version.

Comment (scoder) [reply setupinfo.py]: > from the caller, raising an exception

Yes, that's what I meant. There's probably a place higher up where raising such an exception would be expected.

Comment (hughmcmaster) [reply setupinfo.py]: Okay. I've updated the MR.

**Comment by hughmcmaster** [comment]
Hi there, just checking in on this.

**Comment by scoder** [comment]
Thanks

**Pull request merged**